internal/ais/api.go

```diff
@@ -0,0 +1,61 @@
+package ais
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/google/uuid"
+	temporalsdk_client "go.temporal.io/sdk/client"
+)
+
+func NewAPIServer(ctx context.Context, tc temporalsdk_client.Client, listen, tq string) *http.Server {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/stored", func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodPost {
+			http.Error(w, "Invalid request method", http.StatusMethodNotAllowed)
+			return
+		}
+
+		type data struct {
+			UUID string `json:"uuid"`
+			Name string `json:"name"`
+		}
+		var payload data
+		if err := json.NewDecoder(r.Body).Decode(&payload); err != nil {
+			http.Error(w, "Invalid JSON", http.StatusBadRequest)
+			return
+		}
+
+		if payload.Name == "" {
+			http.Error(w, "Invalid AIP name", http.StatusBadRequest)
+			return
+		}
+
+		aipUUID, err := uuid.Parse(payload.UUID)
+		if err != nil {
+			http.Error(w, "Invalid AIP UUID", http.StatusBadRequest)
+			return
+		}
+
+		err = StartWorkflow(ctx, tc, tq, &WorkflowParams{
+			AIPUUID: aipUUID,
+			AIPName: payload.Name,
+		})
+		if err != nil {
+			http.Error(w, "Couldn't start AIS workflow", http.StatusInternalServerError)
+			return
+		}
+
+		w.WriteHeader(http.StatusAccepted)
+	})
```

Comment:
Service callback configuration needed in the AMSS to hit this endpoint, running all locally:

- Event: `Post-store AIP`
- URI: `http://host.docker.internal:9035/stored`
- Method: `POST`
- Headers (key/value): `Content-type` `application/json`
- Body:
  ```
  {
    "uuid": "<package_uuid>",
    "name": "<package_name>"
  }
  ```
- Expected Status: `202`

I'll document this properly if we decide to use this approach in the end.

If you try this locally using AM docker env, you will need to add the following to the `archivematica-storage-service` container in the `docker-compose` file:

```
extra_hosts:
  - "host.docker.internal:host-gateway"
```